php/libraries/Database.class.inc

```diff
@@ -302,7 +305,7 @@ class Database
 
         if (DEBUG) {
             fwrite(STDERR, $query."\n");
-            return;
+            throw new DatabaseException("Exception due to debug mode");
```

Comment:
It'll still print the queries after crashing.

I have to make it, at least, return `false`, because the method signature says it'll return a boolean, which should be `true` on success.